plugin/src/serialize.rs

```diff
@@ -67,34 +64,112 @@ pub fn serialize(name: &TokenTree, stmts: Vec<Stmt>) -> TokenStream {
-            Stmt::PrefixStmt(s)
-            | Stmt::Stmt(s) => {
+            Stmt::GlobalJumpTarget(
+                n,
+                Relocation {
+                    target_offset,
+                    field_offset,
+                    ref_offset,
+                    kind,
+                },
+            ) => (
+                "global_reloc",
+                vec![
+                    expr_string_from_ident(&n),
+                    target_offset,
+                    Literal::u8_suffixed(field_offset).into(),
+                    Literal::u8_suffixed(ref_offset).into(),
+                    kind,
+                ],
+            ),
+            Stmt::ForwardJumpTarget(
+                n,
+                Relocation {
+                    target_offset,
+                    field_offset,
+                    ref_offset,
+                    kind,
+                },
+            ) => (
+                "forward_reloc",
+                vec![
+                    expr_string_from_ident(&n),
+                    target_offset,
+                    Literal::u8_suffixed(field_offset).into(),
+                    Literal::u8_suffixed(ref_offset).into(),
+                    kind,
+                ],
+            ),
+            Stmt::BackwardJumpTarget(
+                n,
+                Relocation {
+                    target_offset,
+                    field_offset,
+                    ref_offset,
+                    kind,
+                },
+            ) => (
+                "backward_reloc",
+                vec![
+                    expr_string_from_ident(&n),
+                    target_offset,
+                    Literal::u8_suffixed(field_offset).into(),
+                    Literal::u8_suffixed(ref_offset).into(),
+                    kind,
+                ],
+            ),
+            Stmt::DynamicJumpTarget(
+                expr,
+                Relocation {
+                    target_offset,
+                    field_offset,
+                    ref_offset,
+                    kind,
+                },
+            ) => (
+                "dynamic_reloc",
+                vec![
+                    expr,
+                    target_offset,
+                    Literal::u8_suffixed(field_offset).into(),
+                    Literal::u8_suffixed(ref_offset).into(),
+                    kind,
+                ],
+            ),
+            Stmt::BareJumpTarget(
+                expr,
+                Relocation {
+                    field_offset,
+                    ref_offset,
+                    kind,
+                    ..
+                },
+            ) => (
+                "bare_reloc",
+                vec![
+                    expr,
+                    Literal::u8_suffixed(field_offset).into(),
+                    Literal::u8_suffixed(ref_offset).into(),
+                    kind,
+                ],
+            ),
+            Stmt::PrefixStmt(s) | Stmt::Stmt(s) => {
```

Comment:
I'm torn on this one, the original had quite long lines but this feels a bit overkill as well.